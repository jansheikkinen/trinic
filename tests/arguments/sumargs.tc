// Sum Arguments

sum TestSum
  A(T),
  B(T, V),
  C(int8),
  Z
end
