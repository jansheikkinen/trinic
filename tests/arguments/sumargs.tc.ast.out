## PARSER ##
Parsing tests/arguments/sumargs.tc...

### LEXER ###
Generating tokens from source...
Generated 21 tokens
000:          SUM,     (null), (003, 003)
001:   IDENTIFIER,    TestSum, (003, 011)
002:   IDENTIFIER,          A, (004, 003)
003:            (,     (null), (004, 004)
004:   IDENTIFIER,          T, (004, 005)
005:            ),     (null), (004, 006)
006:            ,,     (null), (004, 007)
007:   IDENTIFIER,          B, (005, 003)
008:            (,     (null), (005, 004)
009:   IDENTIFIER,          T, (005, 005)
010:            ,,     (null), (005, 006)
011:   IDENTIFIER,          V, (005, 008)
012:            ),     (null), (005, 009)
013:            ,,     (null), (005, 010)
014:   IDENTIFIER,          C, (006, 003)
015:            (,     (null), (006, 004)
016:         INT8,     (null), (006, 008)
017:            ),     (null), (006, 009)
018:            ,,     (null), (006, 010)
019:   IDENTIFIER,          Z, (007, 003)
020:          END,     (null), (008, 003)

### AST GENERATION ###
sum TestSum A(T,   ), B(T, V,   ), C(INT8,   ), Z,    end

